### PR TITLE
ROX-25358: rename `enabled` feature flags option to `released`

### PR DESCRIPTION
**Pull request by 0x656b694d**
### Description

Let's enforce enabled = released with the naming.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing

- [x] inspected CI results

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [x] modified existing tests

#### How I validated my change

Unit tests.

Current dependencies on/for this PR:

* [master](../tree/master)
  * **PR #12158**
    * **PR #12159** 👈
    * **PR #12122**

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ type FeatureFlag interface {
 	Name() string
 	EnvVar() string
 	Enabled() bool
-	Default() bool
+	Released() bool
```
Comment (janisz) [inline pkg/features/features.go:18]: Maybe not in this PR but could you add comments to describe what's the difference between `Released` and `Enabled`

File: pkg/features/options.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func withUnchangeable(unchangeable bool) option {
 }
 
 var (
-	enabled            = withEnabledByDefault() // Deprecated: just remove the flag.
+	released           = withReleased() // Deprecated: just remove the flag.
```
Comment (janisz) [inline pkg/features/options.go:27]: Wait, do we really need this flag if it was already marked as deprecated (although not correctly)?

Comment (0x656b694d) [reply pkg/features/options.go:27]: * #12170

**Comment by RTann** [comment]
Is this the right ticket reference? The referenced ticket seems to talk about `infractl`

File: pkg/features/features.go (hunk 1 above)
Comment (0x656b694d) [reply pkg/features/features.go:18]: #12122

**Pull request closed without merging**